components/brave_wallet/browser/solana_tx_manager.cc

```diff
@@ -85,19 +89,197 @@ void SolanaTxManager::AddUnapprovedTransaction(
     return;
   }
 
-  SolanaTxMeta meta(from, std::move(tx));
-  meta.set_id(TxMeta::GenerateMetaID());
-  meta.set_origin(
+  auto meta = std::make_unique<SolanaTxMeta>(from, std::move(tx));
+  meta->set_id(TxMeta::GenerateMetaID());
+  meta->set_origin(
       origin.value_or(url::Origin::Create(GURL("chrome://wallet"))));
-  meta.set_created_time(base::Time::Now());
-  meta.set_status(mojom::TransactionStatus::Unapproved);
-  meta.set_chain_id(chain_id);
-  if (!tx_state_manager_->AddOrUpdateTx(meta)) {
+  meta->set_created_time(base::Time::Now());
+  meta->set_status(mojom::TransactionStatus::Unapproved);
+  meta->set_chain_id(chain_id);
+
+  // If the transaction is partially signed, we can't modify the instructions
+  // to add a priority fee.
+  if (meta->tx()->IsPartialSigned()) {
+    if (!tx_state_manager_->AddOrUpdateTx(*meta)) {
+      std::move(callback).Run(
+          false, "", l10n_util::GetStringUTF8(IDS_WALLET_INTERNAL_ERROR));
+      return;
+    }
+    std::move(callback).Run(true, meta->id(), "");
+    return;
+  }
+
+  // Fetch the compute units used for priority fee estimation
+  // by using the simulateTransaction API.
+  auto internal_callback =
+      base::BindOnce(&SolanaTxManager::ContinueAddUnapprovedTransaction,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+
+  GetSolanaTxFeeEstimationForMeta(chain_id, std::move(meta),
+                                  std::move(internal_callback));
+}
+
+void SolanaTxManager::OnSimulateSolanaTransaction(
+    const std::string& chain_id,
+    std::unique_ptr<SolanaTxMeta> meta,
+    uint64_t base_fee,
+    GetSolanaTxFeeEstimationForMetaCallback callback,
+    uint64_t compute_units_consumed,
+    mojom::SolanaProviderError error,
+    const std::string& error_message) {
+  if (error != mojom::SolanaProviderError::kSuccess) {
+    // If the simulation fails, we'll still propagate the base
+    // fee - the client can use it even if the priority fee fails.
+    mojom::SolanaFeeEstimationPtr estimation =
+        mojom::SolanaFeeEstimation::New();
+    estimation->base_fee = base_fee;
+    estimation->compute_units = 0;
+    estimation->fee_per_compute_unit = 0;
+    std::move(callback).Run(std::move(meta), std::move(estimation), error,
+                            error_message);
+    return;
+  }
+
+  // Fetch the fee estimate
```

Comment:
I think no need for this comment, it's fetching recent priority fee and the method name below is clear stating that.